here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -50,6 +50,7 @@ public class ExtensionCache {
       new ConcurrentHashMap<>();
   private static final Map<String, FileClient> jarClientMap = new HashMap<>();
   private final @NotNull INaksha naksha;
+  private Map<String, IExtensionInit> instanceCache = new ConcurrentHashMap<>();
```

Comment:
having another map (instanceCache) in addition to loaderCache breaks the atomicity of operations.
I recommend using loaderCache itself for storing whatever we want to cache against extensionId.
loaderCache can continue to be a Map, but value can be a new class ValueTuple<Extension, ClassLoader, IExtensionInit> instead of existing KVPair<Extension, ClassLoader>...
This will simplify cache operations as we use only one cache.

Reply:
Done.